experimental_design/assay_simulator.py

```diff
@@ -0,0 +1,279 @@
+        titration profiles will be drawn.
+    t_equil: int
+        The index from which it the assaytools PyMC MCMC samples are equilibrated. i.e. the end of the burnin-period.
+
+    Returns
+    -------
+    (cv, var, bias): (np.ndarray, np.ndarray, np.ndarray)
+        the mean coefficien of variation (cv), variance (var) and bias of the fitted free energy for each protein
+        concentration.
+
+    Example
+    -------
+    Run the assaytools PyMC pipeline and save the MCMC samples, here called 'pymc_data'. It's worth discarding the
+    initial samples where the MCMC appears unequilibrated. The end of the burn-in time can be estimated with
+    pymbar.timeseries by looking at trace of the free energy:
+    >>> from pymbar import timeseries
+    >>> (t_equil, g, N_eff) = timeseries.detectEquilibration(pymc_data['DeltaG'][0], fast=True, nskip=1)
+    We'll be using t_equil below.
+
+    We can now estimate how the protein concentration used in the assay will affect the coefficient of variation,
+    variance, and bias of the fitted free energy.
+
+    First, choose the ligand concentrations that can be used in the assay:
+    >>> L_total = 10 ** (np.linspace(-10, -5, num=12))
+
+    Next, pick the range of protein concentrations for which the error of the binding free energy will be estimated:
+    >>> P_totals =10 ** (np.linspace(-10, -1, num=20))
+
+    Estimating the variance error meaures for the above concentrations. Note how we're inputing the burn-in time, as
+    well parsing the assay parameters.
+    >>> (CV, Var, Bias2) = predict_assay_error(pymc_data, L_total, P_totals, t_equil=t_equil, geometry='top',
+    >>>                                        assay_volume=100E-6, well_area=0.3969)
+
+    Finally, it's a question of picking the protein concentration that reduces whichever metric you prefer.
+    """
+    # Pre-assigning the coefficient of variation, the bias, and variance respectively.
+    CV = []
+    Bias2 = []
+    Var = []
+
+    delta_gs = pymc_data['DeltaG'][0]           # All of the binding free energy samples
+    mean_delta_g = np.mean(delta_gs)            # Used to determine the coefficient of variation.
+    pymc_indices = range(t_equil, len(delta_gs))     # The indices of the PyMC samples that are after the burn-in period:
+
+
+    # Looping over protein concentrations, drawing from the posterior and fitting the affinity:
+    for p in range(len(p_total)):
+        estimates = []
+        bias_squared = []
+        # Draw parameters from the posterior
+        for i in range(nposterior_samples):
+            ind = np.random.choice(pymc_indices,1)[0]
+            simulator = AssaySimulator(pymc_data=pymc_data, l_total=l_total, sample_index=ind,  p_total=p_total[p], **kwargs)
+            # Draw fluorescence data with different values of random noise
+            for j in range(nsamples):
+                fit = simulator.fit_deltaG()
```

Comment:
Since `fit_deltaG` currently doesn't add any noise, won't this omit the uncertainty in the measurements?